JavaFiddle-ejb/src/main/java/ru/javafiddle/core/ejb/CompileAndRunBean.java

```diff
@@ -1,27 +1,200 @@
+        try {
+            super.init(parentLoader);
+        } catch (Exception e) {
+            LOG.error("Exception", e);
+        }
+    }
+
     //!TODO
-    public String compile(String projectHash){
-        return null;
+    public String compile(String projectHash) {
+        ByteArrayOutputStream baos = new ByteArrayOutputStream();
+        PrintStream printStream = new PrintStream(baos);
+        PrintStream old = System.out;
+        System.setOut(printStream);
+        BasicConfigurator.configure();
+        List<File> sources;
+        FileBean fileBean = new FileBean();
+        sources = fileBean.getProjectFiles(projectHash);
+        try {
+            init(ClassLoader.getSystemClassLoader());
+        } catch (NullPointerException e) {
+            LOG.error("NullPointerException in getSystemClassLoader()", e);
+        }
+        List<SimpleJavaFileObject> userSources = new ArrayList<>();
+        for (int i = 0; i < sources.size(); i++) {
+            File file = sources.get(i);
+            // typeId = 8 - "class", typeId = 9 - "interface", typeId = 10 - "exception"
+            // typeId = 5 - "runnable", typeId = 11 - "enum", typeId = 12 - "annotation"
+            int typeId = file.getType().getTypeId();
+            if (typeId == 8 || typeId == 9 || typeId == 10 || typeId == 5 || typeId == 11 || typeId == 12) {
+                byte[] userFByte = file.getData();
+                String userFile = "";
+                for (int j = 0; j < userFByte.length; j++) {
+                    userFile += (char) userFByte[j];
+                }
+                String s = findSrc(sources);
+                String src = file.getPath();
+                src = src.replaceFirst(s + "/", "");
+                src = src.replace("/", ".");
+                int pos = src.lastIndexOf(file.getFileName());
+                String pack = src.substring(0, pos);
+                String className = file.getFileName() + ".java";
+                SimpleJavaFileObject sjf = null;
+                if (pack != null) {
+                    sjf = counstructResource(className, userFile, pack);
+                } else {
+                    sjf = counstructResource(className, userFile);
+                }
+                userSources.add(sjf);
+            }
+        }
+
+        String message = "";
+
+        try {
+            message = compileToClass(userSources);
+        } catch (Exception e) {
+            LOG.error("Exception, can't compile", e);
+        }
+        LOG.info(message);
+        System.out.flush();
+        System.setOut(old);
+        return baos.toString();
     }
 
     //!TODO
-    public String run(String projectHash){
-        return null;
+    public String run(String projectHash) {
+        List<File> sources;
+        FileBean fileBean = new FileBean();
+        sources = fileBean.getProjectFiles(projectHash);
+        String mes = "";
+        if (!getMessage().equals("")) {
```

Comment:
Bad style. First of all, it's not null-safe.
Then, Apache StringUtils.isEmpty() looks much better for that.